src/patterns.jl

```diff
@@ -0,0 +1,63 @@
+
+"""
+    Pattern
+
+A pattern is just a wrapper around a `Tuple{Vararg{Symbol}}` which enables searching for
+matching components and dimension paths in a Dataset.
+Special symbols `:_` and `:__` are used as wildcards, similar to `*` and `**` in glob
+pattern matching.
+"""
```

Comment:
it would be helpful to demonstrate a simple filtering example (like you do in the tests but pared back for simplicity).

Reply:
Added a jldoctest with a couple reduced examples.